### ISSUE TITLE
Add in custom sections

### ISSUE THREAD

**Issue by jeeva86** [saadq/resumake.io#97]
**Describe the solution you'd like**
Ability to add custom sections. e.g. Volunteer Work, Summary, Objectives, etc.

**Describe alternatives you've considered**
There's no alternative for Volunteer Work.

**Comment by MatheusRV** [comment]
I can handle with that. @jeeva86 any chance to you help?

**Comment by jeeva86** [comment]
Sure, I can help @MatheusRV

**Comment by MatheusRV** [comment]
Lets start mapping whats have to be done. Which fields do you think would be useful? What should each contain?

Help me make clear what should be done, I believe that I implement this very quickly.

**Comment by jeeva86** [comment]
Some use cases:

1. `Volunteer Work` heading with fields identical to `Work Experience`.
2. `Summary` heading with one field to add a paragraph of text.

**Comment by saadq** [comment]
I have some stuff to say about this, will do so later tonight!

**Comment by raviriley** [comment]
@saadq any update on this issue?

**Comment by nrgapple** [comment]
Is this one a branch?

**Comment by raviriley** [comment]
@nrgapple doesn't look like it is in development right now

**Comment by saadq** [comment]
Sorry for not updating guys...

I'll have a more official statement on this later, but the tldr is that I have been working on a new version of Resumake from scratch (in a private repo) that will support custom sections with a couple of other people and its been kind of slow due to me starting full-time work and just working on this when I have time.

My big project at work is finishing up, so I believe after this week I'll have a lot more time to work on this. I'll make a post next week about the current status of the new project and I'll see if I can just move it to this repo and make it public so others can contribute too.

**Comment by raviriley** [comment]
@saadq Sounds awesome, I'm sure people would be happy to contribute if you did that. 
I'm actually glad that wasn't added yet, it forced me to learn LaTeX to create my customized version of the template, so I gained a new skill from it!

**Comment by nrgapple** [comment]
@raviriley Would you mind sharing some resources on your LaTeX template?

**Comment by nrgapple** [comment]
@saadq And to piggy back this issue, are you continuing to develop in react? Do you have any plans to support [FRESH resume schema](https://github.com/fresh-standard/fresh-resume-schema)? Please don't remove the JSON import support. It's a killer feature! 

From what I can find online, this is the best implementation of JSON Resume on the web. Keep up the great work!

**Comment by raviriley** [comment]
> @raviriley Would you mind sharing some resources on your LaTeX template?

You can check it out [here](https://www.overleaf.com/read/yjqwjrxvrtxz).

**Comment by saadq** [comment]
Closing in favor of [custom sections](https://github.com/saadq/resumake.io/issues/97) which has been implemented in `v3` (hopefully releasing in a few weeks).

**Comment by saadq** [comment]
This has been implemented in `v3` (hopefully releasing it in a few weeks).

**Comment by saadq** [comment]
^

![lie](https://user-images.githubusercontent.com/5678694/133712566-8d7423f0-6d88-49ca-8c41-347a88b28df3.png)

I'm hoping to go back to working on v3 in a couple of weeks when things slow down at work... I will reopen this until v3 is actually published and users can actually use custom sections.

**Comment by carun** [comment]
Hey @saadq, just touching base to see if there was any progress on this :)

**Comment by saadq** [comment]
Hi @carun , if there are any updates/progress I'll be commenting on this issue. It's basically where I left it:

https://user-images.githubusercontent.com/5678694/140445052-294d2644-0721-4a0b-a741-9b52fa96016f.mp4

Right now, I've been working on a different project in my spare time, and hope to have that done within 1-2 weeks. Once that is launched, I will be getting back to working on Resumake.

---

In case anyone is interested what I'm working on, It's basically a centralized portal to track JavaScript language changes/new features.


https://user-images.githubusercontent.com/5678694/140445998-828a8d7d-ba86-4698-9210-6c0d9a6155b3.mp4

**Comment by saadq** [comment]
Published https://proposals.es now, so I think it's time to start working on Resumake again, probably from next weekend.

**Comment by JaxSoder** [comment]
@saadq is there any update on this? If not I would be willing to take over this

**Comment by jeeva86** [comment]
hey all, it's me again, coming back after 3 years haha, surprised that no progress on this still? I even tried to hack the json to put in a new section but that didn't work. I'm trying to use the awards section to highlight volunteer experience now...

**Comment by deduble** [comment]
Not trying to be too obnoxious but isn't there really a progress on this? Seems almost like a couple of lines of code.

**Comment by saadq** [comment]
Hi all,

I'm going to address this and give an update for everyone tomorrow (dw, it's not a project cancellation announcement or something like that)

**Comment by deduble** [comment]
interesting the repo owner wrote something and then removed it

**Comment by saadq** [comment]
Yeah sorry, 2 days ago I wrote "I'll try to give an update tomorrow" but then I realized I wouldn't be able to so I removed it. Going to give it now though :).

---

>  coming back after 3 years haha, surprised that no progress on this still?

Just to give people context, when I initially created this website I was a student and had a lot of free time. I work fulltime now (as of ~3 years ago), and so instead of dedicating large amounts of a week to work on it, I had very small windows where I was able to make updates in my free time (mostly weekends). Because of that, progress had declined drastically. The project was never abandoned, but it was hard to dedicate a lot of time to work on it after I had a fulltime job. Additionally, the fact that I was losing money every month to keep the website running also hurt my motivation and I had been considering just taking down the website and maybe just leaving the source code there for others to fork it and host their own version.

> Not trying to be too obnoxious but isn't there really a progress on this? Seems almost like a couple of lines of code.

Dw, I don't think you're trying to be obnoxious, but if it was just having to do a couple of lines of code I would have done it long ago 😅

The demo I showed above had a few issues and was only working for that one template. Allowing custom sections still requires significant rewrites for the client, the server, and updating every LaTeX template themselves to have generalized stuff I can use in different types of custom sections.

---

OK so with that aside, I wanted to give an update (and I'm probably going to make a more general GitHub issue about this later) - I am stepping away from open source for awhile and wanted to thank you all for using Resumake and appreciate all the good things you've said about it. I do apologize for not being able to ship these features.

With that said, the good news is I'm leaving the responsibility of the project to a friend of mine (@devdrowsy) who is willing to take over the project for some time. I had briefed him on the v3 status, the direction I wanted to go in, and what features to add (some that you are aware of and some cool new ones that should be a nice surprise). I am looking forward to seeing how the project grows while he's maintaining it. 

I've also stressed that I would like for the project to be worked on incrementally instead of shipping large rewrites all at once. Once the core functionality of v3 is done, the plan is to create a separate v3 beta site that is going to be updated more frequently. This will allow people to use v3 a lot sooner (even if it doesn't look as nice or has all the new features done).

If you're in the Resumake discord, you would already know that he was actually going to take over a couple of months ago but he then asked for a break to focus on exams before he took on the role. I'm expecting him to start giving updates some time next week and have asked him to post here when he does so (or maybe in a separate GitHub issue). I've also asked him to reach out to @ducaale if you're still interested in contributing going forward as well :D

Apologies again and thanks everyone for your patience! <3

**Comment by ducaale** [comment]
Thanks for the update @saadq. I will be looking forward to getting in touch with @devdrowsy soon. 

>Additionally, the fact that I was losing money every month to keep the website running also hurt my motivation and I had been considering just taking down the website and maybe just leaving the source code there for others to fork it and host their own version.

I'm sad to hear this. I was hoping that the LaTex generation could be moved to the client-side by utilizing WebAssembly or converting the project into an electron app to reduce the server costs.

**Comment by deduble** [comment]
@saadq I totally feel you about full-time work burdens and good luck on that. I think this is a great choice of you that you let someone else take it instead of very possibly "letting it die". Thanks for all your work on this repo.
